Marlin/planner.h

```diff
@@ -393,7 +393,7 @@ class Planner {
 
     #if ENABLED(ENSURE_SMOOTH_MOVES)
       static bool long_move() {
-          if (blocks_queued()) {
+          if (blocks_queued() && block_buffer_runtime_us) {
```

Comment:
Thanks for the hint.
When this line appeared first it looked like:
```
+        if (blocks_queued()) {
+          block_t* block = &block_buffer[block_buffer_tail];
+          return (block->segment_time > (LCD_UPDATE_THRESHOLD * 1000UL));
+        }
+        else
+          return true;
+      }
```
If there is a __valid__ block then compare time of the last block.

Now we have a readily calculated summ of all blocks, always available.

```
      static bool long_move() {
          if (block_buffer_runtime_us) {
            return block_buffer_runtime_us > (LCD_UPDATE_THRESHOLD) * 1000UL + (MIN_BLOCK_TIME) * 3000UL;
          }
          else
            return true;
      }
```
If there is a `block_buffer_runtime_us` different from 0 there must be blocks.
There can also be a valid block when `block_buffer_runtime_us` is zero, but that is the special case where this block is already running.